### PR TITLE
QEMU command system refactoring

### PR DESCRIPTION
**Pull request by rmalmain**
added the possibility to add custom command definitions

### CODE REVIEW

File: fuzzers/qemu_systemmode/src/fuzzer_sync_exit.rs [hunk 1]
```diff
@@ -52,11 +53,15 @@ pub fn fuzz() {
...
+
         // let emu_snapshot_manager = QemuSnapshotBuilder::new(true);
         let emu_snapshot_manager = FastSnapshotManager::new(false); // Create a snapshot manager (normal or fast for now).
         let emu_exit_handler: StdEmulatorExitHandler<FastSnapshotManager> =
             StdEmulatorExitHandler::new(emu_snapshot_manager); // Create an exit handler: it is the entity taking the decision of what should be done when QEMU returns.
-        let emu = Emulator::new(&args, &env, emu_exit_handler).unwrap(); // Create the emulator
+
+        let cmd_manager = StdCommandManager::new();
+
+        let emu = Emulator::new(&args, &env, emu_exit_handler, cmd_manager).unwrap(); // Create the emulator
```
Comment (domenukk) [inline fuzzers/qemu_systemmode/src/fuzzer_sync_exit.rs:64]: Would a builder make more sense here? What's the `CommandManager`'s use case?

File: libafl_qemu/src/command/mod.rs [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+        InputVirtCommandParser,
+        SaveCommandParser,
+        LoadCommandParser,
+        EndCommandParser,
+        VersionCommandParser,
+        VaddrFilterAllowRangeCommandParser
+    ]
+);
+
+pub trait CommandManager<E, QT, S>: Sized
```
Comment (domenukk) [inline libafl_qemu/src/command/mod.rs:157]: Related question, when would I implement my own `CommandManager`?

File: fuzzers/qemu_systemmode/src/fuzzer_sync_exit.rs (hunk 1 above)
Comment (rmalmain) [reply fuzzers/qemu_systemmode/src/fuzzer_sync_exit.rs:64]: yes, I am coincidentally writing the builder for `Emulator`. i'll add it to this PR when it's done.

File: libafl_qemu/src/command/mod.rs (hunk 2 above)
Comment (rmalmain) [reply libafl_qemu/src/command/mod.rs:157]: It's useful to add custom or optional commands (for example to add some commands depending on a feature being turned on) or for compatibility with other command systems (we will need it at some points to make libafl_qemu compatible with kAFL/Nyx targets for example).

The old system relies on a huge match, it makes it quite inflexible and unreadable for something different from the main usecase.

This traits makes possible something [like this](https://github.com/AFLplusplus/LibAFL/blob/53a5832ac7cb06bf2941be1bf5b372c3ff17614f/libafl_qemu/src/command/mod.rs#L142-L155), which is much cleaner imho.

**Review by domenukk — APPROVED** [review]
Feel free to merge :)

**Pull request merged**